pkg/auth/identity.go

```diff
@@ -18,39 +18,63 @@ type Identity struct {
 	Client  k8sclient.Client
 }
 
+func (r *Identity) CanI(namespace, resource, verb string) (bool, error) {
+	sar := authapi.SelfSubjectAccessReview{
+		Spec: authapi.SelfSubjectAccessReviewSpec{
+			ResourceAttributes: &authapi.ResourceAttributes{
+				Resource:  resource,
+				Namespace: namespace,
+				Verb:      verb,
+			},
+		},
+	}
+	err := r.Client.Create(context.TODO(), &sar)
+	if err != nil {
+		return false, err
+	}
+	return sar.Status.Allowed, nil
+}
+
 func (r *Identity) HasRead(namespaces []string) (Authorized, error) {
 	authorized := Authorized{}
 	err := r.BuildClient()
 	if err != nil {
 		return authorized, err
 	}
 	for _, namespace := range namespaces {
-		sar := authapi.SelfSubjectAccessReview{
-			Spec: authapi.SelfSubjectAccessReviewSpec{
-				ResourceAttributes: &authapi.ResourceAttributes{
-					Resource:  "namespaces",
-					Namespace: namespace,
-					Verb:      "get",
-				},
-			},
-		}
-		err := r.Client.Create(context.TODO(), &sar)
+		allowed, err := r.CanI(namespace, "namespaces", "get")
 		if err != nil {
 			return authorized, err
 		}
-		authorized[namespace] = sar.Status.Allowed
+		authorized[namespace] = allowed
 	}
 
 	return authorized, nil
 }
 
 func (r *Identity) HasMigrate(namespaces []string) (Authorized, error) {
+	migrateResources := []string{"pods", "deployments", "deploymentconfigs", "daemonsets", "replicasets", "statefulsets", "pvcs"}
+
 	authorized := Authorized{}
 	err := r.BuildClient()
 	if err != nil {
 		return authorized, err
 	}
 
+	for _, namespace := range namespaces {
+		authorized[namespace] = true
+		for _, resource := range migrateResources {
+			allowed, err := r.CanI(namespace, resource, "*")
```

Comment:
This may actually not be sufficient. We may end up doing a lot of individual SAR checks here and that's okay.

See:
```
[dymurray@pups mig-controller]$ oc auth can-i '*' deployments                                                                                                               
no - no RBAC policy matched
[dymurray@pups mig-controller]$ oc auth can-i update deployments 
yes
```

I think we need to do individual verb checks on these resources:
`get`, `create`, `update`.